lib/autoload/course/assessment/programming_test_case_report.rb

```diff
@@ -171,6 +202,22 @@ def passed?
       !failed? && !skipped? && !errored?
     end
 
+    # Combines the different strings above into a hash
+    #
+    # @return [Hash]
+    def messages
+      @messages ||= {
+        'error': error_message,
+        'error_contents': error_contents,
+        'hint': hint,
+        'failure': failure_message,
+        'failure_contents': failure_contents,
+        'output': output
+      }
+      # prune empty and nil values
+      @messages.reject! { |_, v| v.blank? }
```

Comment:
This should be directly applied to the hash, otherwise everytime we call `messages`, the `reject` is called, not really cached.